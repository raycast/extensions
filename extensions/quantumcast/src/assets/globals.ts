import { cloudflowBaseUrl, userHomeDir } from "./preferences";

/**
 * Quantumcast Documentation globals
 */
export const docBaseUrl = "https://jcgerhard.gitbook.io/quantumcast";
export const docUrlGoToCollection = `${docBaseUrl}/manual/commands/go-to-collection`;
export const docUrlGoToWhitepaper = `${docBaseUrl}/manual/commands/go-to-whitepaper`;
export const docUrlGoToWorkspace = `${docBaseUrl}/manual/commands/go-to-workspace`;
export const docUrlGoToFolder = `${docBaseUrl}/manual/commands/go-to-folder`;
export const docUrlGoToFile = `${docBaseUrl}/manual/commands/go-to-file`;
export const docUrlCreateCustomObject = `${docBaseUrl}/manual/commands/create-custom-object`;
export const docUrlOpenAppBuilder = `${docBaseUrl}/manual/commands/open-appbuilder`;

/**
 * Cloudflow globals
 */
export const cloudflowCollectionsUrl = `${cloudflowBaseUrl}/mongoCloud/mongoCloud_nebula.html?collection=`;
export const cloudflowWhitepaperUrl = `${cloudflowBaseUrl}/portal.cgi?quantum&whitepaperName=`;
export const cloudflowAssetUrl = `${cloudflowBaseUrl}/portal.cgi?asset=`;
export const cloudflowAppBuilderUrl = `${cloudflowBaseUrl}/appnavigator/index.html?cfappName=`;
export const cloudflowAppDetailsUrl = `${cloudflowBaseUrl}/appsnebula/index.html#/details`;
export const cloudflowApiJsonUrl = `${cloudflowBaseUrl}/portal.cgi?api=json`;
export const cloudflowApiDocUrl = `${cloudflowBaseUrl}/?api=getJavaScriptDocumentation`;
export const cloudflowApiWebServiceUrl = `${cloudflowBaseUrl}/portal.cgi?api=getWebServiceDocumentation`;
export const cloudflowManualUrl = `${cloudflowBaseUrl}/manual/manual.html`;
export const cloudflowProofscopeUrl = `${cloudflowBaseUrl}/portal.cgi?proofscope&topbar=true&url=`;
export const cloudflowProofscopeFileTypes = [".pdf", ".jpg", ".bmp", ".cf2"];
//export const cloudflowWhereAreTheNodes = `/Applications/Cloudflow/220902_cloudflow_mac_lts_b45297/workers/Resource/WhereAreTheNodes.json`

/**
 * Packz globals
 */
export const packzPresetsBaseURL = "/Users/Shared/PACKZ Software/";
export const packzPresetsUrlUser = `${packzPresetsBaseURL}/User`;
export const packzPresetsUrlShared = `${packzPresetsBaseURL}/Shared`;
export const packzPresetsUrlCloudflow = `${packzPresetsBaseURL}/Cloudflow`;
export const packzFileTypes = [".pdf", ".ai", ".tiff", ".tif", ".jpg", ".png", ".psd", ".eps", ".ic3d"];

/**
 * Frame (Plugin-Suite) globals
 */
// export const framePresetsUrl = `${userHomeDir}/Documents/NiXPS/FrameWorkstationSetup.json`;
// New location from 22.12 on...
export const framePresetsUrl = `${userHomeDir}/Library/Application Support/Hybrid/FrameWorkstationSetup.json`;

/**
 * TRAC globals
 */
export const tracBaseUrl = "https://trac.hybridsoftware.com/hybrid";
export const tracQueryUrl = `${tracBaseUrl}/query`;
