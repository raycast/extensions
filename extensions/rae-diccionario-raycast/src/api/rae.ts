import fetch from "node-fetch";

const API_BASE_URL = "https://rae-api.com";

export interface WordMeaning {
  meaning_number: number;
  category: string;
  description: string;
}

export interface Word {
  word: string;
}

export interface WordEntry extends Word {
  meanings: {
    senses: WordMeaning[];
  }[];
}

export interface ApiResponse<T = unknown> {
  ok: boolean;
  data: T;
  error?: string;
}

export type WordOnlyResponse = ApiResponse<Word>;

export type WordEntryResponse = ApiResponse<WordEntry>;

// Helper function to make API requests and handle errors
async function makeApiRequest<T>(url: string): Promise<T> {
  const response = await fetch(url);

  if (!response.ok) {
    throw new Error(`Error en la petici√≥n: ${response.statusText}`);
  }

  const res = (await response.json()) as ApiResponse<T>;

  if (!res.ok) {
    // Handle the common error structure: {"error":"NOT_FOUND","ok":false}
    const errorMsg = res.error === "NOT_FOUND" ? "Palabra no encontrada" : res.error;
    throw new Error(`Error en la respuesta de la API: ${errorMsg}`);
  }

  return res.data;
}

export const searchWord = async (word: string): Promise<WordEntry> => {
  return makeApiRequest<WordEntry>(`${API_BASE_URL}/api/words/${encodeURIComponent(word)}`);
};

export const getDailyWord = async (): Promise<WordEntry> => {
  const res = await makeApiRequest<Word>(`${API_BASE_URL}/api/daily`);
  return searchWord(res.word);
};

export const getRandomWord = async (minLength?: number, maxLength?: number): Promise<WordEntry> => {
  let url = `${API_BASE_URL}/api/random`;
  const params = new URLSearchParams();

  if (minLength) params.append("min_length", minLength.toString());
  if (maxLength) params.append("max_length", maxLength.toString());

  if (params.toString()) {
    url += `?${params.toString()}`;
  }

  const res = await makeApiRequest<Word>(url);
  return searchWord(res.word);
};
