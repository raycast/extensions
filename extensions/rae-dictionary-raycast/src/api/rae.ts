import fetch from "node-fetch";

const API_BASE_URL = "https://rae-api.com";

// Definition of a word (senses)
export interface Definition {
  raw: string;
  meaning_number: number;
  category: string;
  verb_category?: string;
  gender?: string;
  article?: Article;
  usage: string;
  description: string;
  synonyms: string[];
  antonyms: string[];
}

export interface Article {
  category: string;
  gender: string;
}

export interface Origin {
  raw: string;
  type: string;
  voice: string;
  text: string;
}

export interface Conjugation {
  singular_first_person: string;
  singular_second_person: string;
  singular_formal_second_person: string;
  singular_third_person: string;
  plural_first_person: string;
  plural_second_person: string;
  plural_formal_second_person: string;
  plural_third_person: string;
}

export interface ConjugationNonPersonal {
  infinitive: string;
  participle: string;
  gerund: string;
  compound_infinitive: string;
  compound_gerund: string;
}

export interface ConjugationIndicative {
  present: Conjugation;
  present_perfect: Conjugation;
  imperfect: Conjugation;
  past_perfect: Conjugation;
  preterite: Conjugation;
  past_anterior: Conjugation;
  future: Conjugation;
  future_perfect: Conjugation;
  conditional: Conjugation;
  conditional_perfect: Conjugation;
}

export interface ConjugationSubjunctive {
  present: Conjugation;
  present_perfect: Conjugation;
  imperfect: Conjugation;
  past_perfect: Conjugation;
  future: Conjugation;
  future_perfect: Conjugation;
}

export interface ConjugationImperative {
  singular_second_person: string;
  singular_formal_second_person: string;
  plural_second_person: string;
  plural_formal_second_person: string;
}

export interface Conjugations {
  non_personal: ConjugationNonPersonal;
  indicative: ConjugationIndicative;
  subjunctive: ConjugationSubjunctive;
  imperative: ConjugationImperative;
}

export interface Meaning {
  origin?: Origin;
  senses: Definition[];
  conjugations?: Conjugations;
}

export interface WordEntry {
  word: string;
  meanings: Meaning[];
}

export interface Word {
  word: string;
}

export interface ApiResponse<T = unknown> {
  ok: boolean;
  data: T;
  error?: string;
}

export type WordOnlyResponse = ApiResponse<Word>;
export type WordEntryResponse = ApiResponse<WordEntry>;

// Helper function to make API requests and handle errors
async function makeApiRequest<T>(url: string): Promise<T> {
  const response = await fetch(url);

  if (!response.ok) {
    throw new Error(`Request error: ${response.statusText}`);
  }

  const res = (await response.json()) as ApiResponse<T>;

  if (!res.ok) {
    const errorMsg = res.error === "NOT_FOUND" ? "Word not found" : res.error;
    throw new Error(`API response error: ${errorMsg}`);
  }

  return res.data;
}

export const searchWord = async (word: string): Promise<WordEntry> => {
  return makeApiRequest<WordEntry>(`${API_BASE_URL}/api/words/${encodeURIComponent(word)}`);
};

export const getDailyWord = async (): Promise<WordEntry> => {
  const res = await makeApiRequest<Word>(`${API_BASE_URL}/api/daily`);
  return searchWord(res.word);
};

export const getRandomWord = async (minLength?: number, maxLength?: number): Promise<WordEntry> => {
  let url = `${API_BASE_URL}/api/random`;
  const params = new URLSearchParams();

  if (minLength) params.append("min_length", minLength.toString());
  if (maxLength) params.append("max_length", maxLength.toString());

  if (params.toString()) {
    url += `?${params.toString()}`;
  }

  const res = await makeApiRequest<Word>(url);
  return searchWord(res.word);
};
