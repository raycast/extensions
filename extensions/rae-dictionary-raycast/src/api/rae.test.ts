import { jest } from "@jest/globals";
import { searchWord, getDailyWord, getRandomWord } from "./rae";

// Set longer timeout for API calls
jest.setTimeout(10000);

// Tests are skipped to avoid hitting the real API during CI/CD
describe.skip("RAE API", () => {
  // Real API tests using snapshots
  describe("searchWord", () => {
    it("should fetch and return a known word", async () => {
      // Test with a common word that's unlikely to change in the dictionary
      const result = await searchWord("hola");

      // Basic structure validation
      expect(result.word).toBe("hola");
      expect(result.meanings.length).toBeGreaterThan(0);
      expect(result.meanings[0].senses.length).toBeGreaterThan(0);

      expect(result).toMatchSnapshot({});
    });

    it("should throw error for non-existent words", async () => {
      // Using a gibberish word that shouldn't exist in the dictionary
      await expect(searchWord("xyzxyzxyznotaword")).rejects.toThrow("Request error: Not Found");
    });
  });

  describe("getDailyWord", () => {
    it("should fetch today's word and return its data", async () => {
      const result = await getDailyWord();

      // Basic structure validation
      expect(result.word).toBeTruthy();
      expect(result.meanings.length).toBeGreaterThan(0);

      // Create snapshot of the structure but not the specific word (which changes daily)
      expect(result).toMatchSnapshot({
        word: expect.any(String),
        meanings: expect.any(Array),
      });
    });
  });

  describe("getRandomWord", () => {
    it("should fetch a random word with specified length constraints", async () => {
      const result = await getRandomWord(4, 6);

      // Validate the length constraint
      expect(result.word.length).toBeGreaterThanOrEqual(4);
      expect(result.word.length).toBeLessThanOrEqual(6);

      // Basic structure validation
      expect(result.meanings.length).toBeGreaterThan(0);

      // Create snapshot of the structure but not the specific word (which is random)
      expect(result).toMatchSnapshot({
        word: expect.any(String),
        meanings: expect.any(Array),
      });
    });

    it("should fetch a random word without length constraints", async () => {
      const result = await getRandomWord();

      // Basic structure validation
      expect(result.word).toBeTruthy();
      expect(result.meanings.length).toBeGreaterThan(0);

      // Create snapshot of the structure but not the specific word (which is random)
      expect(result).toMatchSnapshot({
        word: expect.any(String),
        meanings: expect.any(Array),
      });
    });
  });
});
