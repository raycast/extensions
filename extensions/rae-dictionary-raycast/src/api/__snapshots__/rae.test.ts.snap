// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RAE API getDailyWord should fetch today's word and return its data 1`] = `
{
  "meanings": Any<Array>,
  "word": Any<String>,
}
`;

exports[`RAE API getRandomWord should fetch a random word with specified length constraints 1`] = `
{
  "meanings": Any<Array>,
  "word": Any<String>,
}
`;

exports[`RAE API getRandomWord should fetch a random word without length constraints 1`] = `
{
  "meanings": Any<Array>,
  "word": Any<String>,
}
`;

exports[`RAE API searchWord should fetch and return a known word 1`] = `
{
  "meanings": [
    {
      "origin": {
        "raw": "Voz expr.; cf. ingl. hello, al. hallo.",
        "text": "",
        "type": "",
        "voice": "expressive",
      },
      "senses": [
        {
          "antonyms": null,
          "category": "interjection",
          "description": "como salutaci칩n familiar",
          "meaning_number": 1,
          "raw": "1. interj. U. como salutaci칩n familiar.",
          "synonyms": null,
          "usage": "",
        },
        {
          "antonyms": null,
          "category": "interjection",
          "description": "para denotar extra침eza, placentera o desagradable",
          "meaning_number": 2,
          "raw": "2. interj. p. us. U. para denotar extra침eza, placentera o desagradable. U. t. repetida.",
          "synonyms": null,
          "usage": "rare",
        },
        {
          "antonyms": null,
          "category": "interjection",
          "description": "para llamar a los inferiores",
          "meaning_number": 3,
          "raw": "3. interj. desus. Era u. para llamar a los inferiores.",
          "synonyms": null,
          "usage": "obsolete",
        },
      ],
    },
  ],
  "word": "hola",
}
`;
